### PR TITLE
Add option to customize images used for lambda execution

### PR DESCRIPTION
**Pull request by dominikschubert**
Adds a `RuntimeImageResolver` class and a corresponding new config variable `LAMBDA_RUNTIME_IMAGE_MAPPING`

Copied doc from the class:

> Resolves Lambda runtimes to corresponding docker images
> The default behavior resolves based on a prefix (including the repository) and a suffix (per runtime).
> 
> This can be customized via the LAMBDA_RUNTIME_IMAGE_MAPPING config in 2 distinct ways:
> 
>   Option A: use a pattern string for the config variable that includes the "<runtime>" string
>       e.g. "myrepo/lambda:<runtime>-custom" would resolve the runtime "python3.9" to "myrepo/lambda:python3.9-custom"
> 
>   Option B: use a JSON dict string for the config variable, mapping the runtime to the full image name & tag
>       e.g. {"python3.9": "myrepo/lambda:python3.9-custom", "python3.8": "myotherrepo/pylambda:3.8"}
>   
>       Note that with Option B this will only apply to the runtimes included in the dict.
>       All other (non-included) runtimes will fall back to the default behavior.

sample configurations:

* `LAMBDA_RUNTIME_IMAGE_MAPPING="customrepo/lambda-<runtime>:2022"`
* `LAMBDA_RUNTIME_IMAGE_MAPPING='{"python3.9": "customrepo/lambda-py-thon:3.9"}'`

### CODE REVIEW

File: localstack/services/awslambda/invocation/docker_runtime_executor.py [hunk 1]
```diff
@@ -181,7 +248,7 @@ def start(self, env_vars: dict[str, str]) -> None:
 
     def stop(self) -> None:
         CONTAINER_CLIENT.stop_container(container_name=self.id, timeout=5)
-        CONTAINER_CLIENT.remove_container(container_name=self.id)
+        # CONTAINER_CLIENT.remove_container(container_name=self.id)
```
Comment (whummer) [inline localstack/services/awslambda/invocation/docker_runtime_executor.py:251]: nit: Is this change intended? (as it seems unrelated to the PR)

**Review by whummer — APPROVED** [review]
LGTM 👍 - thanks for jumping on this one! 🚀

**Review by thrau — APPROVED** [review]
LGTM! nice clean config + implementation

File: localstack/services/awslambda/invocation/docker_runtime_executor.py [hunk 2]
```diff
@@ -29,7 +29,7 @@
 
 LOG = logging.getLogger(__name__)
 
-RUNTIME_REGEX = r"(?P<runtime>[a-z]+)(?P<version>\d+(\.\d+)?(\.al2)?)(?:.*)"
+RUNTIME_REGEX = r"(?P<runtime>[a-z]+)(?P<version>\d+(\.\d+)?(\.al2)?)(?:.*)"  # TODO: any reason this is still here?
```
Comment (dfangl) [inline localstack/services/awslambda/invocation/docker_runtime_executor.py:32]: I don't think so.

**Review by dfangl — APPROVED** [review]
LGTM!

File: localstack/services/awslambda/invocation/docker_runtime_executor.py (hunk 1 above)
Comment (dominikschubert) [reply localstack/services/awslambda/invocation/docker_runtime_executor.py]: good catch, this was indeed just a leftover :grin:

File: localstack/services/awslambda/invocation/docker_runtime_executor.py (hunk 2 above)
Comment (dominikschubert) [reply localstack/services/awslambda/invocation/docker_runtime_executor.py]: :broom:

**Review by dfangl — APPROVED** [review]
